src/commandHandler.ts

```diff
@@ -0,0 +1,25 @@
+import { Interaction } from "discord.js";
+import Commands from "./Commands";
+
+export default async(interaction: Interaction): Promise<void> => {
```

Comment:
Maybe only accept the chatInput command type.

Reply:
> Maybe only accept the chatInput command type.

I'm not quite sure what you mean here. Could you provide any more info on what you mean by `chatInput`?

The reason I opted for passing the entire `Interaction` object is it provides us access to both the command that was invoked, as well as any options that were passed into the command.

Whilst the `Interaction` object also contains the entire `Discord.js` client too (among many other things), I didn't see any need for them.

Reply:
I meant the ChatInputInteraction type from DiscordJS since this interaction is specifically a chat input, if paired with guarding.